public/app/features/explore/AddToDashboard/AddToDashboardModal.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { useState } from 'react';
+import { DataQuery, SelectableValue } from '@grafana/data';
+import { locationService } from '@grafana/runtime';
+import { Button, Field, Input, InputControl, Modal, RadioButtonGroup } from '@grafana/ui';
+import { DashboardPicker } from 'app/core/components/editors/DashboardPicker';
+import { FolderPicker } from 'app/core/components/Select/FolderPicker';
+import { DeepMap, FieldError, useForm } from 'react-hook-form';
+import { addToDashboard, SaveToExistingDashboardDTO, SaveToNewDashboardDTO } from './addToDashboard';
+// import { notifyApp } from 'app/core/actions';
+// import { createErrorNotification, createSuccessNotification } from 'app/core/copy/appNotification';
+// import { useDispatch } from 'react-redux';
```

Comment:
this (and related comments in this file) is still WIP, need to check with UX the best way to provide users feedback upon success/failure